docs/notebooks/active_learning.pct.py

```diff
@@ -0,0 +1,188 @@
+
+# %%
+import trieste
+
+observer = trieste.objectives.utils.mk_observer(scaled_branin)
+
+num_initial_points = 2
+initial_query_points = search_space.sample_halton(num_initial_points)
+initial_data = observer(initial_query_points)
+
+
+# %% [markdown]
+# ## Surrogate model
+#
+# Just like in sequential optimization, we fit a surrogate Gaussian process model to the initial data.
+
+# %%
+import gpflow
+from trieste.models.gpflow import GPflowModelConfig
+
+
+def build_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.RBF(variance=variance, lengthscales=[2, 2])
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return GPflowModelConfig(
+        **{
+            "model": gpr,
+            "optimizer": gpflow.optimizers.Scipy(),
+            "optimizer_args": {
+                "minimize_args": {"options": dict(maxiter=100)},
+            },
+        }
+    )
+
+
+model = build_model(initial_data)
+
+# %% [markdown]
+# ## Active learning using predictive variance
+#
+# For our first active learning example, we will use a simple acquisition function known as `PredictiveVariance` which chooses points for which we are highly uncertain (i.e. the predictive posterior covariance matrix at these points has large determinant).
+#
+# We will now demonstrate how to choose individual query points using `PredictiveVariance` before moving onto batch active learning. For both cases, we can utilize trieste's `BayesianOptimizer` to do the active learning steps.
+#
+
+# %%
+from trieste.acquisition.optimizer import generate_continuous_optimizer
+from trieste.acquisition.rule import EfficientGlobalOptimization
+from trieste.acquisition.function import PredictiveVariance
+
+acq = PredictiveVariance()
+rule = EfficientGlobalOptimization(
+    builder=acq, optimizer=generate_continuous_optimizer(sigmoid=False)
+)
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+
+# %% [markdown]
+# To plot the contour of variance of our model at each step, we can set the `track_state` parameter to `True` in `bo.optimize()`, this will make trieste record our model at each iteration.
+
+# %%
+bo_iter = 5
+result = bo.optimize(bo_iter, initial_data, model, rule, track_state=True)
+
+# %% [markdown]
+# Then we can retrieve our final dataset from the active learning steps
+
+# %%
+dataset = result.try_get_final_dataset()
+query_points = dataset.query_points.numpy()
+observations = dataset.observations.numpy()
+
+# %% [markdown]
+# Finally, we can check the performance of our `PredictiveVariance` active learning acquisition function by plotting the predictive variance landscape of our model. We can see how it samples regions for which our model is highly uncertain.
+
+# %%
+from util.plotting import plot_bo_points, plot_function_2d
+
+for i in range(bo_iter):
```

Comment:
you could make this a function, since you use exactly the same code below to plot batch results